### PR TITLE
doc(install): Improve installation docs, esp. for Docker

### LINKED ISSUE #169: Installation instructions need significant overhaul
I'm trying to install TRAM but the installation instructions leave a lot to be desired. 

Simply put the docker compose file fails immediately because it cannot locate the Dockerfile. If It give that an absolute path then it gets as far as trying to pip install which fails because it can't locate the requirements.txt file. I can install those manually but then I get an error that it appears not to be a python project because it's missing a pyproject.toml file despite that file being in the directory that the python script is looking in.  

Is there a more complete installation guide for this because simply running docker-compose -f docker-compose.yml up like it says in the instructions is not enough.

### PR DESCRIPTION
**Pull request by mehaase**
Fixes #169

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,27 +42,46 @@ to integrate ATT&CK more easily and consistently into their products.
...
+      shows, "CONTAINER ID   IMAGE     COMMAND" on the first line, then it is
+      running. If it says, "cannot connect to Docker daemon," then Docker is not
+      running.
+2. Download docker-compose.yml for TRAM, using [this
+   link](https://raw.githubusercontent.com/center-for-threat-informed-defense/tram/master/docker/docker-compose.yml)
+   or using curl:
 
     ```shell
-    docker-compose -f docker-compose.yml up
+    $ curl -O https://raw.githubusercontent.com/center-for-threat-informed-defense/tram/master/docker/docker-compose.yml
```
Comment (m3mike) [inline README.md:57]: Really great overall, while you are here, can you update the links to use permalinks? Either to the a specific release (https://github.com/center-for-threat-informed-defense/tram/releases/download/v1.1.2/docker-compose.yml) or a github permalink (https://github.com/center-for-threat-informed-defense/tram/blob/b7268eb5333696ae753d58e7dbb1860293823767/docker/docker-compose.yml for example), so if we reorg the repo or something the links won't break?

File: docker/README.md [hunk 2]
```diff
@@ -1,75 +1,60 @@
...
+      <th>Variable</th>
+      <th>Required<th>
+      <th>Description</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td><code>ALLOWED_HOSTS</code></td>
+      <td>Yes<td>
+      <td>A list of hostnames that TRAM can be served from.</td>
```
Comment (m3mike) [inline docker/README.md:23]: Just an FYI, Github supports Markdown tables in case it's helpful in the future, but this works too.

**Review by m3mike — CHANGES_REQUESTED** [review]
Just a suggestion to update the links while editing the READMEs, to either be permalinks or point to a specific release.

File: docker/README.md (hunk 2 above)
Comment (mehaase) [reply docker/README.md:23]: I don't think that markdown tables can have multiline cells, so they get really wide when one of the columns contains a paragraph of text. (Other than that I like them.)

**Review by m3mike — APPROVED** [review]

**Pull request merged**